### PR TITLE
Add timestamp to log messages.

### PR DESCRIPTION
**Pull request by darko-poljak**
With this change when timestamping is turned on (**-P, --timestamp** option) then log messages are timestamped with the current local date and time in the format **YYYYMMDDhh24mmss.us**.

Sample log lines:
```
INFO: [20180923203734.433345] 185.203.112.26: Starting configuration run
INFO: [20180923203748.876977] 185.203.112.26: Processing __myline/test
ERROR: [20180923203748.881465] 185.203.112.26: /bin/sh -......
```

And in the parallel or parallel jobs mode:
```
INFO: [4306]: [20180923204032.922092] 185.203.112.26: Starting configuration run
INFO: [4306]: [20180923204047.109844] 185.203.112.26: Processing __myline/test
ERROR: [4306]: [20180923204047.114561] 185.203.112.26: /bin/sh...
```

Debug:
```
VERBOSE: cdist: version 4.10.3-1-g6acf6f64
DEBUG: [20180923204150.561461] inventory: Host '185.203.112.26' not found, skipped
DEBUG: cdist: Base root path for target host "185.203.112.26" is "/tmp/tmpx4hk83db/90cb04f9d8b4a35c89571b1c6fb9927c"
DEBUG: [20180923204150.561812] 185.203.112.26: remote_exec for host "185.203.112.26": ssh -o User=root -o ControlPath=/tmp/tmphcvyzfqv/s  -o ControlMaster=auto  -o ControlPersist=2h
DEBUG: [20180923204150.561941] 185.203.112.26: remote_copy for host "185.203.112.26": scp -o User=root -q -o ControlPath=/tmp/tmphcvyzfqv/s  -o ControlMaster=auto  -o ControlPersist=2h
DEBUG: [20180923204150.562029] 185.203.112.26: address family: 0
DEBUG: [20180923204150.566076] 185.203.112.26: derived host_name for host "185.203.112.26": 26.112.203.185.in-addr.arpa
DEBUG: [20180923204150.568094] 185.203.112.26: derived host_fqdn for host "185.203.112.26": 26.112.203.185.in-addr.arpa
DEBUG: [20180923204150.568233] 185.203.112.26: target_host for host "185.203.112.26": ('185.203.112.26', '26.112.203.185.in-addr.arpa', '26.112.203.185.in-addr.arpa')
INFO: [20180923204150.568800] 185.203.112.26: Starting configuration run
DEBUG: [20180923204150.569237] 185.203.112.26: Checking conf_dir /usr/home/darko/ungleich/cdist/cdist/conf ...
DEBUG: [20180923204150.574617] 185.203.112.26: Checking conf_dir /home/darko/.cdist ...
DEBUG: [20180923204150.575587] 185.203.112.26: Checking conf_dir /home/darko/ungleich/dot-cdist ...
DEBUG: [20180923204150.580846] 185.203.112.26: Checking conf_dir /home/darko/ungleich/dot-cdist ...
VERBOSE: [20180923204151.479842] 185.203.112.26: Running global explorers
DEBUG: [20180923204156.133850] 185.203.112.26: Running global explorers sequentially
```

### CODE REVIEW

**Comment by darko-poljak** [comment]
I also fixed getting loggers. getLogger("cdist") was first called before setting logging class depending on args so subsequent calls to getLogger("cdist") returns already created logger.

**Comment by darko-poljak** [comment]
Full sample debug output follows:
```
VERBOSE: cdist: version 4.10.3-2-g0c57b308
DEBUG: [20180923213844.970838] inventory: Host '185.203.112.26' not found, skipped
DEBUG: [20180923213844.970978] config: Base root path for target host "185.203.112.26" is "/tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c"
DEBUG: [20180923213844.971140] 185.203.112.26: remote_exec for host "185.203.112.26": ssh -o User=root -o ControlPath=/tmp/tmp2drn7p10/s  -o ControlMaster=auto  -o ControlPersist=2h
DEBUG: [20180923213844.971200] 185.203.112.26: remote_copy for host "185.203.112.26": scp -o User=root -q -o ControlPath=/tmp/tmp2drn7p10/s  -o ControlMaster=auto  -o ControlPersist=2h
DEBUG: [20180923213844.971254] 185.203.112.26: address family: 0
DEBUG: [20180923213844.974824] 185.203.112.26: derived host_name for host "185.203.112.26": 26.112.203.185.in-addr.arpa
DEBUG: [20180923213844.976660] 185.203.112.26: derived host_fqdn for host "185.203.112.26": 26.112.203.185.in-addr.arpa
DEBUG: [20180923213844.976730] 185.203.112.26: target_host for host "185.203.112.26": ('185.203.112.26', '26.112.203.185.in-addr.arpa', '26.112.203.185.in-addr.arpa')
INFO: [20180923213844.977103] 185.203.112.26: Starting configuration run
DEBUG: [20180923213844.977411] 185.203.112.26: Checking conf_dir /usr/home/darko/ungleich/cdist/cdist/conf ...
DEBUG: [20180923213844.990212] 185.203.112.26: Checking conf_dir /home/darko/.cdist ...
DEBUG: [20180923213844.991166] 185.203.112.26: Checking conf_dir /home/darko/ungleich/dot-cdist ...
DEBUG: [20180923213844.997854] 185.203.112.26: Checking conf_dir /home/darko/ungleich/dot-cdist ...
VERBOSE: [20180923213845.880550] 185.203.112.26: Running global explorers
DEBUG: [20180923213850.527850] 185.203.112.26: Running global explorers sequentially
DEBUG: [20180923213850.528186] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213850.642131] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213850.743154] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213850.838083] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213850.934382] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.031893] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.128882] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.225368] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.322345] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.419781] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.516296] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.609819] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.706216] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.806877] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213851.902342] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213852.003648] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213852.098264] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213852.196589] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213852.291978] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213852.386142] 185.203.112.26: return_output is True, ignoring stdout
VERBOSE: [20180923213852.482130] 185.203.112.26: Running initial manifest /home/darko/.cdist/manifest/init-j
DEBUG: [20180923213852.482476] 185.203.112.26: /home/darko/.cdist/manifest/init-j is NOT executable, running it with /bin/sh -e
DEBUG: [20180923213854.920047] 185.203.112.26: Iteration in sequential mode
VERBOSE: [20180923213854.920646] 185.203.112.26: Preparing object __file/root/host.file
VERBOSE: [20180923213854.920711] 185.203.112.26: Running manifest and explorers for __file/root/host.file
VERBOSE: [20180923213854.920769] 185.203.112.26: Running type explorers for <CdistType __file>
DEBUG: [20180923213856.363964] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213856.460533] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213856.561081] 185.203.112.26: return_output is True, ignoring stdout
VERBOSE: [20180923213856.658771] 185.203.112.26: Running object __file/root/host.file
DEBUG: [20180923213856.658995] 185.203.112.26: Generating code for __file/root/host.file
DEBUG: [20180923213856.659511] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__file/gencode-local is executable, running it
DEBUG: [20180923213856.665517] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__file/gencode-remote is executable, running it
VERBOSE: [20180923213856.670301] 185.203.112.26: Preparing object __file/root/host.file3
VERBOSE: [20180923213856.670402] 185.203.112.26: Running manifest and explorers for __file/root/host.file3
VERBOSE: [20180923213856.670468] 185.203.112.26: Running type explorers for <CdistType __file>
DEBUG: [20180923213857.207940] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213857.304696] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213857.404262] 185.203.112.26: return_output is True, ignoring stdout
VERBOSE: [20180923213857.499277] 185.203.112.26: Running object __file/root/host.file3
DEBUG: [20180923213857.499495] 185.203.112.26: Generating code for __file/root/host.file3
DEBUG: [20180923213857.506525] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__file/gencode-local is executable, running it
DEBUG: [20180923213857.512988] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__file/gencode-remote is executable, running it
VERBOSE: [20180923213857.517603] 185.203.112.26: Preparing object __file/root/test.pipe
VERBOSE: [20180923213857.517708] 185.203.112.26: Running manifest and explorers for __file/root/test.pipe
VERBOSE: [20180923213857.517772] 185.203.112.26: Running type explorers for <CdistType __file>
DEBUG: [20180923213858.054919] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213858.150429] 185.203.112.26: return_output is True, ignoring stdout
DEBUG: [20180923213858.250269] 185.203.112.26: return_output is True, ignoring stdout
VERBOSE: [20180923213858.344968] 185.203.112.26: Running object __file/root/test.pipe
DEBUG: [20180923213858.345113] 185.203.112.26: Generating code for __file/root/test.pipe
DEBUG: [20180923213858.345357] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__file/gencode-local is executable, running it
DEBUG: [20180923213858.351802] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__file/gencode-remote is executable, running it
VERBOSE: [20180923213858.356977] 185.203.112.26: Preparing object __myline/test
VERBOSE: [20180923213858.357126] 185.203.112.26: Running manifest and explorers for __myline/test
VERBOSE: [20180923213858.357188] 185.203.112.26: Running type explorers for <CdistType __myline>
DEBUG: [20180923213859.304102] 185.203.112.26: return_output is True, ignoring stdout
VERBOSE: [20180923213859.400759] 185.203.112.26: Running type manifest /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__myline/manifest for object __myline/test
DEBUG: [20180923213859.401125] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__myline/manifest is NOT executable, running it with /bin/sh -e
VERBOSE: [20180923213859.404146] 185.203.112.26: Running object __myline/test
DEBUG: [20180923213859.404302] 185.203.112.26: Generating code for __myline/test
DEBUG: [20180923213859.404553] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__myline/gencode-local is NOT executable, running it with /bin/sh -e
DEBUG: [20180923213859.407168] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/conf/type/__myline/gencode-remote is executable, running it
INFO: [20180923213859.413487] 185.203.112.26: Processing __myline/test
DEBUG: [20180923213859.413848] 185.203.112.26: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/object/__myline/test/.cdist-s5lqvyve/code-local is NOT executable, running it with /bin/sh -e
ERROR: [20180923213859.418225] 185.203.112.26: /bin/sh -e /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/object/__myline/test/.cdist-s5lqvyve/code-local: ['/bin/sh', '-e', '/tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/object/__myline/test/.cdist-s5lqvyve/code-local']

Error processing object '__myline/test'
=======================================
name: __myline/test
path: /tmp/tmprsmrtnwe/90cb04f9d8b4a35c89571b1c6fb9927c/data/object/__myline/test/.cdist-s5lqvyve
source: /home/darko/.cdist/manifest/init-j
type: /usr/home/darko/.cdist/type/__myline

manifest:stdout
---------------
myline manifest stdout

manifest:stderr
---------------
myline manifest stderr

code-local:stdout
-----------------
something to stdout

code-local:stderr
-----------------
error

gencode-remote:stderr
---------------------
test gencode-remote error


VERBOSE: [20180923213859.418440] config: Total processing time for 1 host(s): 14.45188021659851
ERROR: cdist: Failed to configure the following hosts: 185.203.112.26

```

File: cdist/argparse.py [hunk 1]
```diff
@@ -251,6 +251,11 @@ def get_parsers():
                   'line). If no host or host file is specified then, by '
                   'default, read hosts from stdin.'),
             dest='hostfile', required=False)
+    parser['config_args'].add_argument(
+           '-P', '--timestamp',
```
Comment (telmich) [inline cdist/argparse.py]: somebody suggested -t like strace -- did we already use that one?

**Comment by telmich** [comment]
Otherwise good!

**Pull request merged**